src/authorization/jwt/JwtValidator.ts

```diff
@@ -4,55 +4,88 @@
 import * as ccfapp from "@microsoft/ccf-app";
 import { ServiceResult } from "../../utils/ServiceResult";
 import { IValidatorService } from "../IValidationService";
-import { JwtIdentityProviderEnum } from "./JwtIdentityProviderEnum";
-import { IJwtIdentityProvider } from "./IJwtIdentityProvider";
-import { DemoJwtProvider } from "./DemoJwtProvider";
-import { MsJwtProvider } from "./MsJwtProvider";
 import { Logger, LogContext } from "../../utils/Logger";
+import { JwtValidationPolicyMap } from "./JwtValidationPolicyMap";
 
 export class JwtValidator implements IValidatorService {
-  private readonly identityProviders = new Map<
-    JwtIdentityProviderEnum,
-    IJwtIdentityProvider
-  >();
   private logContext: LogContext;
 
   constructor(logContext?: LogContext) {
     this.logContext = (logContext?.clone() || new LogContext()).appendScope("JwtValidator");
-    this.identityProviders.set(
-      JwtIdentityProviderEnum.MS_AAD,
-      new MsJwtProvider("JwtProvider", this.logContext),
-    );
-    this.identityProviders.set(
-      JwtIdentityProviderEnum.Demo,
-      new DemoJwtProvider("DemoJwtProvider"),
-    );
   }
 
   validate(request: ccfapp.Request<any>): ServiceResult<string> {
     const jwtCaller = request.caller as unknown as ccfapp.JwtAuthnIdentity;
     Logger.debug(
-      `Authorization: JWT jwtCaller (JwtValidator)-> ${<JwtIdentityProviderEnum>jwtCaller.jwt.keyIssuer}`,
+      `Authorization: JWT jwtCaller (JwtValidator)-> ${jwtCaller.jwt.keyIssuer}`,
       this.logContext
     );
-    const provider = this.identityProviders.get(
-      <JwtIdentityProviderEnum>jwtCaller.jwt.keyIssuer,
-    );
-
-    if (!provider) {
-      const error = `Authorization: JWT validation provider is undefined (JwtValidator) for ${<JwtIdentityProviderEnum>jwtCaller.jwt.keyIssuer}`;
-      Logger.error(error, this.logContext);
+    const issuer = jwtCaller?.jwt?.payload?.iss;
+    if (!issuer) {
       return ServiceResult.Failed(
-        { errorMessage: error, errorType: "caller error" },
+        {
+          errorMessage: "The JWT has no valid iss",
+          errorType: "AuthenticationError",
+        },
         400,
         this.logContext
       );
     }
-    const isValidJwtToken = provider.isValidJwtToken(jwtCaller);
+
```

Comment:
reminder: we need to check if issuer is in our accepted issuers table.

Reply:
I don't believe this code is functionally different at all from the previous code, so if this code is missing something, the original code is also missing something